server/resolvers/TestResultOperations/saveTestResultCommon.js

```diff
@@ -92,8 +98,30 @@ const saveTestResultCommon = async ({
 
     await updateTestPlanRun(testPlanRun.id, { testResults: newTestResults });
 
-    // TODO: Avoid blocking loads in test runs with a larger amount of tests
-    //       and/or test results
+    if (isSubmit) {
```

Comment:
Adding this block to address an easily reproducible process breakage where the metrics of a TestPlanReport could fail to be tracked.

Scenario:
1. Tester is assigned to a Test Run on Test Queue
2. They go through the tests for the assigned run in their Test Run page, submitting results along the way
3. Returns to the Test Queue page
4. Reselects their assigned Test Run
5. Selects "Edit Results"
6. Makes a change to the test assertion that's collected or unexpected behaviors
7. Selects "Submit Results" (they never navigated to another test in the Test Run)

The expectation is that the metrics would be recalculated in the instance of a "Success case" going to a "Failure case". But this doesn't happen. It seems the metrics were only being recorded on test transition within the test run. This change makes sure it is also done when those results are submitted.